audit/src/EthPlaysV0.sol

```diff
@@ -0,0 +1,480 @@
+        if (
+            block.timestamp <
+            controlAuctionStartTimestamp + controlAuctionDuration
+        ) {
+            revert AuctionInProgress();
+        }
+
+        if (bestControlBid.from == address(0)) {
+            revert AuctionHasNoBids();
+        }
+
+        emit Control(bestControlBid.from);
+        controlAddress = bestControlBid.from;
+        bestControlBid = ControlBid(address(0), 0);
+        controlAuctionEndTimestamp = block.timestamp;
+    }
+
+    /* -------------------------------------------------------------------------- */
+    /*                                   ADMIN                                    */
+    /* -------------------------------------------------------------------------- */
+
+    /// @notice Set the isActive parameter. Owner only.
+    /// @param _isActive New value for the isActive parameter
+    function setIsActive(bool _isActive) external onlyOwner {
+        isActive = _isActive;
+        emit SetIsActive(_isActive);
+    }
+
+    function setAlignmentDecayRate(uint256 _alignmentDecayRate)
+        external
+        onlyOwner
+    {
+        alignmentDecayRate = _alignmentDecayRate;
+        emit SetAlignmentDecayRate(_alignmentDecayRate);
+    }
+
+    function setChaosVoteReward(uint256 _chaosVoteReward) external onlyOwner {
+        chaosVoteReward = _chaosVoteReward;
+        emit SetChaosVoteReward(_chaosVoteReward);
+    }
+
+    function setOrderDuration(uint256 _orderDuration) external onlyOwner {
+        orderDuration = _orderDuration;
+        emit SetOrderDuration(_orderDuration);
+    }
+
+    function setChaosInputRewardCooldown(uint256 _chaosInputRewardCooldown)
+        external
+        onlyOwner
+    {
+        chaosInputRewardCooldown = _chaosInputRewardCooldown;
+        emit SetChaosInputRewardCooldown(_chaosInputRewardCooldown);
+    }
+
+    function setChaosInputReward(uint256 _chaosInputReward) external onlyOwner {
+        chaosInputReward = _chaosInputReward;
+        emit SetChaosInputReward(_chaosInputReward);
+    }
+
+    function setOrderInputReward(uint256 _orderInputReward) external onlyOwner {
+        orderInputReward = _orderInputReward;
+        emit SetOrderInputReward(_orderInputReward);
+    }
+
+    function setChatCost(uint256 _chatCost) external onlyOwner {
+        chatCost = _chatCost;
+        emit SetChatCost(_chatCost);
+    }
+
+    function setRareCandyCost(uint256 _rareCandyCost) external onlyOwner {
+        rareCandyCost = _rareCandyCost;
+        emit SetRareCandyCost(_rareCandyCost);
+    }
+
+    function setControlAuctionDuration(uint256 _controlAuctionDuration)
+        external
+        onlyOwner
+    {
+        controlAuctionDuration = _controlAuctionDuration;
+        emit SetControlAuctionDuration(_controlAuctionDuration);
+    }
+
+    function setControlDuration(uint256 _controlDuration) external onlyOwner {
+        controlDuration = _controlDuration;
+        emit SetControlDuration(_controlDuration);
+    }
+}
```

Comment:
Do you expect these values to change a lot? Wondering about a bunch of mutable values like you have here vs. deploying a new contract with new constants (both in terms of code complexity but also gas reading from each config variable in the game logic above). Not sure what chain you're deploying to so trade-offs will be different per chain.

Reply:
I don't expect them to change much. This will be deployed to the "cheap chain", which will be Polygon for the first round. So, deployment costs are very low - mostly concerned about gas costs of the game functions above.

I think they should be in this contract to avoid the external function call overhead. However, I wonder if the code complexity issue could be addressed by storing all of the configuration variables in a single struct (then only one setter function required)? That would also allow the admin to update multiple values at once, which would be convenient. Will look into this shortly.